Ataque e movimentação do inimigo

ArthurLimaS opened the pull request:
I) adicionadas as seguintes variaveis na struct Inimigo: dano (int), cor (Color) e framesCounter (int).

II) tipos da variaveis de hp mudados de float para int.

III) adicionado código que faz o inimigo se apróximar do player quando o mesmo se apróxima.

IV) adicionado código para um ataque do inimigo (devido a falta de sprite, o indicador é a mudança de cor do inimigo)

V) adicionado um código que faz a vida do jogador aparecer na tela (apenas o número)

ArthurLimaS committed:
Ataque Inimigo #17 #19

I) adicionadas as seguintes variaveis na struct Inimigo: dano (int), cor (Color) e framesCounter (int).

II) tipos da variaveis de hp mudados de float para int.

III) adicionado código que faz o inimigo se apróximar do player quando o mesmo se apróxima.

IV) adicionado código para um ataque do inimigo (devido a falta de sprite, o indicador é a mudança de cor do inimigo)

V) adicionado um código que faz a vida do jogador aparecer na tela (apenas o número)

ArthurLimaS committed:
Merge branch 'master' into Ataque_Inimigo

VelocityIsntSpeed commented on jogo42-cb/main.c:
```diff
@@ -73,7 +81,41 @@ int main(void)
 
         // Mover inimigo
-        // TODO
+        if(CheckCollisionCircles((Vector2){gs->inim.pos.x, gs->inim.pos.y}, 160, (Vector2){gs->jog.pos.x, gs->jog.pos.y}, RAIO_JOG))
```
`gs->inim.pos` já é um `Vector2`, então `(Vector2){gs->inim.pos.x, gs->inim.pos.y}` tem valor igual e é redundante. O mesmo vale para `gs->jog.pos`.

```suggestion
        if(CheckCollisionCircles(gs->inim.pos, 160, gs->jog.pos, RAIO_JOG))
```

VelocityIsntSpeed commented on jogo42-cb/main.c:
```diff
@@ -73,7 +81,41 @@ int main(void)
+        }
+
+        // Ataque Inimigo
+        if(CheckCollisionCircles((Vector2){gs->inim.pos.x, gs->inim.pos.y}, 40, (Vector2){gs->jog.pos.x, gs->jog.pos.y}, RAIO_JOG))
```
`gs->inim.pos` já é um `Vector2`, então `(Vector2){gs->inim.pos.x, gs->inim.pos.y}` tem valor igual e é redundante. O mesmo vale para `gs->jog.pos`.

```suggestion
        if(CheckCollisionCircles(gs->inim.pos, 40, gs->jog.pos, RAIO_JOG))
```

VelocityIsntSpeed commented on jogo42-cb/main.c:
```diff
@@ -32,25 +33,32 @@ int main(void)
     gs->obstCircCentro = (Vector2){1500, 350};
     gs->obstCircRaio = 150;
     gs->obstCircTaAndando = false;
+    */
```
Isso não desativa os obstáculos, e sim apenas deixa de inicializá-los. Porém esses valores não inicializados ainda estão sendo usados pelo programa, o que causa bugs.

VelocityIsntSpeed replied:
Descomentei com a commit e18d6df671e442fd1b5ba1fa00ef41af97026143

VelocityIsntSpeed commented on jogo42-cb/jogador.c:
```diff
@@ -47,6 +47,12 @@ static bool ColisaoJogLevel(const Vector2 posJogTeste, const GameState* gs)
     }
 
+    //[ CHECAR CONTRA O INIMIGO ]-----------------------------------------------
+    if (CheckCollisionCircles((Vector2){gs->inim.pos.x, gs->inim.pos.y}, 20, posJogTeste, RAIO_JOG))
```
`gs->inim.pos` já é um `Vector2`, então `(Vector2){gs->inim.pos.x, gs->inim.pos.y}` tem valor igual e é redundante.

```suggestion
    if (CheckCollisionCircles(gs->inim.pos, 20, posJogTeste, RAIO_JOG))
```

VelocityIsntSpeed committed:
Aplica sugestões da review

VelocityIsNotSpeed commented:
Referências a outras issues/PRs podem estar em sumários de commits ou em comentários de issues/PRs, mas não em títulos de issues/PRs.

VelocityIsntSpeed committed:
Reativa inicialização dos obstáculos

Ver comentário de review na PR #30

VelocityIsNotSpeed commented:
Vou dar uma reorganizada, mover as coisas para um novo arquivo `inimigo.c`.

VelocityIsntSpeed committed:
Move código do inimigo para o novo arquivo inimigo.c

VelocityIsntSpeed committed:
Remove acento

Na verdade é um `ç`, mas tem problema do mesmo jeito (eu acho). Dps vou colocar isso nos avisos.

VelocityIsntSpeed committed:
Refatora para melhor legibilidade

De acordo com as diretrizes de estilo: #24

VelocityIsntSpeed committed:
Remove inclusão desnecessária de `math.h`

Todos os arquivos .c incluem `jogo42.h`, que inclui `raymath.h`, que inclui `math.h`.

VelocityIsntSpeed committed:
Move o texto do HP para o canto superior esquerdo

O que tinha antes era uma gambiarra (colocar um newline no final do texto dos controles).

VelocityIsntSpeed committed:
Remove números mágicos em DesenharUI

VelocityIsntSpeed committed:
Renomeia DesenharUI para DesenharHpJog

VelocityIsntSpeed committed:
Remove números mágicos

VelocityIsntSpeed committed:
Mod. comentário de documentação de AtaqueDoInimigo

VelocityIsntSpeed committed:
Melhora código do movimento do inimigo

Antes tava usando a constante de velocidade de movimento do jogador.

VelocityIsntSpeed committed:
Remove dano do inimigo do struct Inimigo para uma constante

O dano não muda, então não tem pra quê estar no struct de estado.

VelocityIsntSpeed committed:
Melhora comentários dos membros do struct Inimigo

VelocityIsntSpeed committed:
Melhora movimento inimigo

VelocityIsntSpeed committed:
Remove membro `seguindo` de Inimigo

Não estava sendo utilizado

VelocityIsntSpeed committed:
Pequenas refatorações misc.

VelocityIsntSpeed committed:
Merge branch 'master' into Ataque_Inimigo

VelocityIsntSpeed committed:
Rem. inclusão redundante de raylib.h em inimigo.c

jogo42.h já inclui raylib.h

VelocityIsntSpeed committed:
Adc. indicadores gráficos e mod. ataque

VelocityIsntSpeed committed:
Implementa rotação do inimigo, com o indicador gráfico

VelocityIsntSpeed committed:
Adc. whitespace e comentários em DesenharInimigo

VelocityIsntSpeed committed:
Renomeia constantes relacionadas ao inimigo

Para que todas comecem com `INIM_`
